src/test/scala/mesosphere/marathon/integration/RemoteRegionOffersIntegrationTest.scala

```diff
@@ -0,0 +1,110 @@
+      val app = appProxy(applicationId, "v1", instances = 1, healthCheck = None)
+
+      When("The app is deployed without specifying region")
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created in the default region")
+      result should be(Created)
+      extractDeploymentIds(result)
+      waitForDeployment(result)
+      waitForTasks(app.id.toPath, 1)
+      val slaveId = marathon.tasks(applicationId).value.head.slaveId.get
+      val agentRegion = mesos.state.value.agents.find(_.id == slaveId).get.attributes.attributes("fault_domain_region")
+
+      agentRegion match {
+        case ITResourceStringValue(value) => value shouldEqual homeRegion.value
+      }
+    }
+    "Launch an instance of the app in the specified region" in {
+      val applicationId = appId(Some("must-be-placed-in-remote-region"))
+      val app = appProxy(applicationId, "v1", instances = 1, healthCheck = None).copy(constraints =
+        Set(Constraints.regionField :: "LIKE" :: remoteRegion.value :: Nil))
+
+      When("The app is deployed with specific region constraint")
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created in the specified region")
+      result should be(Created)
+      extractDeploymentIds(result)
+      waitForDeployment(result)
+      waitForTasks(app.id.toPath, 1)
+      val slaveId = marathon.tasks(applicationId).value.head.slaveId.get
+      val agentRegion = mesos.state.value.agents.find(_.id == slaveId).get.attributes.attributes("fault_domain_region")
+
+      agentRegion match {
+        case ITResourceStringValue(value) => value shouldEqual remoteRegion.value
+      }
+    }
+    "Launch an instance of the app in the specified region and zone" in {
+      val applicationId = appId(Some("must-be-placed-in-remote-region"))
+      val app = appProxy(applicationId, "v1", instances = 1, healthCheck = None).copy(constraints = Set(
+        Constraints.regionField :: "LIKE" :: remoteRegion.value :: Nil,
+        Constraints.zoneField :: "LIKE" :: remoteZone2.value :: Nil
+      ))
+
+      When("The app is deployed with specific region and zone constraints")
+      val result = marathon.createAppV2(app)
+
+      Then("The app is created in the proper region and a proper zone")
+      result should be(Created)
+      extractDeploymentIds(result)
+      waitForDeployment(result)
+      waitForTasks(app.id.toPath, 1)
+      val slaveId = marathon.tasks(applicationId).value.head.slaveId.get
+      val agentRegion = mesos.state.value.agents.find(_.id == slaveId).get.attributes.attributes("fault_domain_region")
+      val agentZone = mesos.state.value.agents.find(_.id == slaveId).get.attributes.attributes("fault_domain_zone")
+
+      agentRegion match {
+        case ITResourceStringValue(value) => value shouldEqual remoteRegion.value
+      }
+      agentZone match {
+        case ITResourceStringValue(value) => value shouldEqual remoteZone2.value
+      }
+    }
+  }
+
+}
```

Comment:
Looks like you can get away with 2 agents since you don't use `remoteZone1`?

Reply:
We still have to see that we don't place anything on the other agent

Reply:
But where do you actually check it?

Reply:
```scala
inside(agentZone) {
        case ITResourceStringValue(value) => value shouldEqual remoteZone2.value
      }
```
We check that task was launched on the agent with a zone2 and not other agent.